core/dbt/task/test.py

```diff
@@ -237,6 +237,12 @@ def execute_unit_test(
                 "Invalid materialization context generated, missing config: {}".format(context)
             )
 
+        # allows test code to access the configuration settings of the component it's testing
+        if unit_test_node.tested_node_unique_id:
```

Comment:
@takemikami These commits trace back to a blog post you made. Could you explain the use case behind these lines? After some thoughtful experimentation, I don't see how they are used or what user feature they are enabling. I was able to use the other code in this PR to enable injection of a sql header inside and will have a PR for this soon. However, could you answer what these lines are for? (I tried a few obvious things like {{ tested_node_config }} but this leads to a compiler use before instantiation problem)